packages/smart-contracts/src/contracts/ERC20EscrowToPayV1.sol

```diff
@@ -0,0 +1,385 @@
+     * @param _payee Transfer recipient.
+     * @param _amount Amount to transfer.
+     * @param _feeAmount The amount of fee to pay.
+     * @param _feeAddress The fee recipient.
+     * @dev The fees amount is transfered from the payer's account.
+     * @return success if transaction successeded. 
+     */
+    function openEscrow(
+        bytes memory _paymentRef,
+        IERC20 _paymentToken,
+        uint256 _amount,
+        address _payee,
+        uint256 _feeAmount,
+        address _feeAddress
+    ) 
+        external
+        payable
+        returns (bool success)
+    {
+        require(invoiceMapping[_paymentRef].amount == 0, 
+            "ERC20EscrowToPayV1: This invoice already exists, is this the correct paymentReference?"
+        ); 
+        
+        uint256 sixMonths = 15778458;
+        // claimDate is used to set a timestamp on when the payee can execute closeEscrow.
+        uint256 _claimDate = block.timestamp + sixMonths;
+        
+        invoiceMapping[_paymentRef] = Invoice(
+            _paymentToken, 
+            _amount,
+            _payee,
+            msg.sender,
+            _feeAmount,
+            _feeAddress,
+            _claimDate
+        );
+        
+        require(_deposit(_paymentRef),"OpenEscrow: Call to _deposit failed!");
+
+        emit OpenEscrow(_paymentRef);
+        return true;
+    }
+    
+    /**
+     * @notice Closes an open escrow and pays the invoice request to it's issuer.
+     * @param _paymentRef Reference of the related Invoice.
+     * @dev The fees is paid to the fee address at this point.
+     */
+    function closeEscrow(bytes memory _paymentRef) external {
+        require(invoiceMapping[_paymentRef].amount != 0, "ERC20EscrowToPay: Payment reference does not exist");
+        require(msg.sender == invoiceMapping[_paymentRef].payer ||
+            (msg.sender == invoiceMapping[_paymentRef].payee  &&
+            invoiceMapping[_paymentRef].claimDate <= block.timestamp),
+            "ERC20EscrowToPay: Only the payer or payee can excecute this call!"
+        );
+        require(_withdraw(_paymentRef, invoiceMapping[_paymentRef].payee), "CloseEscrow: Withdraw failed!");
+        
+        delete invoiceMapping[_paymentRef];
+
+        emit EscrowCompleted(_paymentRef);  
+    }
+    
+    /**
+     * @notice Allaws the payer to initiate a dispute if they have an isssue with the payee.
+     * @param _paymentRef Reference of the Invoice related.
+     */
+    function openDispute(bytes memory _paymentRef) external {
+        require(msg.sender == invoiceMapping[_paymentRef].payer, 
+            "ERC20EscrowToPay: You are not the valid payer. Only the valid payer can open a dispute!");
+        require(invoiceMapping[_paymentRef].amount != 0, 
+            "ERC20EscrowToPay: No dispute found, Wrong payment reference?");
+        require(disputeMapping[_paymentRef].amount == 0, "ERC20EscrowToPay: A dispute already exists, is this the correct paymentRef?");
```

Comment:
Same comment to reduce gas usage in error messages

Reply:
```suggestion
        require(disputeMapping[_paymentRef].amount == 0, "ERC20EscrowToPay: A dispute already exist.");
```